src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -140,4 +157,23 @@ private boolean rowBatcherErrorIsKnownServerBug(Exception ex) {
         }
         return false;
     }
+
+    private String buildMaxValueDslQuery(String currentQuery, String constrainingColumn) {
```

Comment:
The complexity of this logic - and my experience with QueryMarkLogic in NiFi, where all of this complexity was stuffed into the QueryMarkLogic class (I've moved some of it out) and thus was fairly difficult to understand - makes me think the refactor I mentioned above will definitely be worth it. And instead of "ConstraintInjector", I think the interface would be for "I can both inject a constraint and also get you the max value for a constraint column", so.... perhaps `ConstraintColumnHandler` for lack of a better verb now (handle = process = "I'm not sure what else to call this"). 

And then I think two methods:

- `injectConstraintIntoQuery`
- `getMaxConstraintValue`

And then we'll have DslQuery and SerializedQuery implementations. 

Again, no need to do this now, as it'll be better to wait to do the serialized query implementation to make sure that the refactor makes sense.

Reply:
Deferred until the serialized constraint story.